sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,230 @@
+
+class EventHubProducerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(
+        cls, conn_str: str,
+        *,
+        event_hub_path: str = None,
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        auth_timeout: float = 60,
+        user_agent: str = None,
+        retry_total: int = 3,
+        transport_type=None) -> EventHubProducerClient:
+
+    async def send(
+        self,
+        event_data: Union[EventData, EventDataBatch, Iterable[EventData]],
+        *,
+        partition_key: Union[str, bytes] = None,
+        partition_id: str = None,
+        timeout: float = None) -> None:
+
+    async def create_batch(self, events: Iterable[EventData] = None, max_size: int = None) -> EventDataBatch:
+
+    async def get_properties(self) -> Dict[str, Any]:
+
+    async def get_partition_ids(self) -> List[str]:
+
+    async def get_partition_properties(self, partition: str) -> Dict[str, str]:
+
+    async def close(self) -> None:
+
+
+class EventHubConsumerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            *
+            partition_manager: PartitionManager,
+            load_balancing_interval: int = 10,
+            logging_enable: bool = False,
+            http_proxy: dict = None,
+            auth_timeout: float = 60,
+            user_agent: str = None,
+            retry_total: int = 3,
+            transport_type=None
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(cls, conn_str: str, **kwargs) -> EventHubConsumerClient:
+
+    async def receive(
+            self,
+            on_event: Callable[PartitionContext, EventData],
+            consumer_group: str,
+            *,
+            partition_id: str = None,
+            owner_level: int = None,
```

Comment:
Need to discuss the locations of:
- `partition_id`
- `owner_level`
- `partition_manager`

On the client? Or on the receive function?

Reply:
for partition_id, on the receive function since you can receive from multiple partitions using the same consumer. At least this is the case in .NET and Java.

Reply:
Sorry - meant to say `consumer_group` not `partition_id`

Reply:
The only argument I can see for having `consumer_group` at the receive function level would be in the case of a single client wanting to operate over multiple consumer groups in order to share the connection.

Reply:
As discussed for JS:
- Moving `consumer_group` to the constructor.
- `owner_level` will stay at the receive function and should be paired with `partition_id`.